solarforecastarbiter/metrics/deterministic.py

```diff
@@ -19,218 +19,276 @@
-    y_norm : float
-        Normalized factor, in the same units as y_true and y_pred.
+    obs : (n,) array-like
+        Observed values.
+    fx : (n,) array-like
+        Forecasted values.
+    norm : float
+        Normalized factor, in the same units as obs and fx.
 
     Returns
     -------
     nrmse : float
-        The NRMSE [%] between the true and predicted values.
+        The NRMSE [%] of the forecast.
 
     """
 
-    return root_mean_square(y_true, y_pred) / y_norm * 100.0
+    return root_mean_square(obs, fx) / norm * 100.0
 
 
-def forecast_skill(y_true, y_pred, y_ref):
+def forecast_skill(obs, fx, ref):
     """Forecast skill (s).
 
+        s = 1 - RMSE_fx / RMSE_ref
+
+    where RMSE_fx is the RMSE of the forecast and RMSE_ref is the RMSE of the
+    reference forecast (e.g. Persistence).
+
     Parameters
     ----------
-    y_true : array-like
-        True values.
-    y_pred : array-like
-        Predicted values.
-    y_ref: array_like
-        Reference forecast values.
+    obs : (n,) array-like
+        Observed values.
+    fx : (n,) array-like
+        Forecasted values.
+    ref : (n,) array_like
+        A reference forecast.
 
     Returns
     -------
     s : float
-        The forecast skill [-] between the true and predicted values compared
-        to a reference forecast.
+        The forecast skill [-] of the forecast relative to a reference
+        forecast.
 
     """
 
-    rmse_pred = root_mean_square(y_true, y_pred)
-    rmse_ref = root_mean_square(y_true, y_ref)
-    return 1.0 - rmse_pred / rmse_ref
+    rmse_fx = root_mean_square(obs, fx)
+    rmse_ref = root_mean_square(obs, ref)
+    return 1.0 - rmse_fx / rmse_ref
 
 
-def pearson_correlation_coeff(y_true, y_pred):
+def pearson_correlation_coeff(obs, fx):
     """Pearson correlation coefficient (r).
 
+        r = A / (B * C)
+
+    where:
+
+        A = sum_{i=1}^n (fx_i - fx_avg) * (obs_i - obs_avg)
+        B = sqrt( sum_{i=1}^n (fx_i - fx_avg)^2 )
+        C = sqrt( sum_{i=1}^n (obs_i - obs_avg)^2 )
+        fx_avg = 1/n sum_{i=1} fx_i
+        obs_avg = 1/n sum_{i=1} obs_i
+
     Parameters
     ----------
-    y_true : array-like
-        True values.
-    y_pred : array-like
-        Predicted values.
+    obs : (n,) array-like
+        Observed values.
+    fx : (n,) array-like
+        Forecasted values.
 
     Returns
     -------
     r : float
-        The correlation coefficient (r [-]) between the true and predicted
-        values.
+        The correlation coefficient (r [-]) of the observations and forecasts.
 
     """
 
-    x1 = y_pred - np.mean(y_pred)
-    x2 = y_true - np.mean(y_true)
-    return np.sum(x1 * x2) / (
-        np.sqrt(np.sum(x1 ** 2)) * np.sqrt(np.sum(x2 ** 2))
-    )
+    fx_avg = np.mean(fx)
+    obs_avg = np.mean(obs)
+    A = np.sum((fx - fx_avg) * (obs - obs_avg))
+    B = np.sqrt(np.sum((fx - fx_avg) ** 2))
+    C = np.sqrt(np.sum((obs - obs_avg) ** 2))
+    r = A / (B * C)
+    return r
```

Comment:
should have asked earlier... why don't we use [`numpy.corrcoef`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.corrcoef.html)?

Reply:
Good idea. (I forgot about `numpy.corrcoef()`.)

I'll revise accordingly.